### PR TITLE
[Proposal] Support saving to safetensors

### PR DESCRIPTION
**Pull request by MatthieuBizien**
Follow-up of https://github.com/huggingface/diffusers/pull/1357, and mimics Transformers https://github.com/huggingface/transformers/pull/20321

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
@Narsil what do you think?

File: tests/test_pipelines.py [hunk 1]
```diff
@@ -508,6 +508,18 @@ def test_set_scheduler_consistency(self):
 
         assert dict(ddim_config) == dict(ddim_config_2)
 
+    def test_save_safe_serialization(self):
+        pipeline = StableDiffusionPipeline.from_pretrained("hf-internal-testing/tiny-stable-diffusion-torch")
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            tmpdirname = "/tmp/test_save_safe_serialization"  # TODO: remove
+            pipeline.save_pretrained(tmpdirname, safe_serialization=True)
```
Comment (Narsil) [inline tests/test_pipelines.py:515]: Should we maybe verify the type of files that have been saved ?

**Review by Narsil — APPROVED** [review]
LGTM. Thanks !

File: tests/test_pipelines.py (hunk 1 above)
Comment (MatthieuBizien) [reply tests/test_pipelines.py:516]: Good point, fixed in 8c12a559c399e87e31f853701e31bb8e634ca7cf

**Review by Narsil — APPROVED** [review]

**Comment by patrickvonplaten** [comment]
Great work @MatthieuBizien - thanks for the PR :-)

**Comment by luiscape** [comment]
Hello @Narsil @patrickvonplaten 👋! Awesome to see this merged.

I am wondering if you have latency figures that compare loading a Stable Diffusion pipeline with and without `safetensors` on GPUs. Documentation elsewhere seems to suggest this should be faster. I am wondering if that's actually the case and, if so, how much faster that is.

**Comment by Narsil** [comment]
It's really hard to answer in general. It will skip a cpu allocation and create objects directly on the gpu (using SAFETENSORS_FAST_GPU=1 environment variable). For cpu the loading part should be almost instant in a lot of cases. But it does depend on hardware.

Here is someone that had lots of success apparently. 

https://gist.github.com/xrpgame/8f756f99b00b02697edcd5eec5202c59?permalink_comment_id=4388611#gistcomment-4388611

Some report barely a change.

Personally it's mostly 2x.
Didn't see anyone observing it being slower though.

So I guess you're going to have to measure😉

**Pull request merged**